### PR TITLE
notmuch: fix cross

### PR DESCRIPTION
**Pull request by ghost**
Adapted from Void Linux commit 81c3c1884be1959337b55cd88d4748d17ac6c3de and f95b944252e3a1fc900a18c8eeed59e6de9b6e9a.

Note: `withEmacs` will be disabled automatically due to
- https://github.com/NixOS/nixpkgs/pull/225123

### CODE REVIEW

File: pkgs/applications/networking/mailreaders/notmuch/default.nix [hunk 1]
```diff
@@ -37,17 +43,39 @@ stdenv.mkDerivation rec {
     pythonPackages.python
   ] ++ lib.optional withRuby ruby;
 
+  patches = [
+    (fetchpatch {
+      url = "https://raw.githubusercontent.com/void-linux/void-packages/master/srcpkgs/notmuch/patches/0001-configure-detect-version-by-compiler-for-build.patch";
```
Comment (flokli) [inline pkgs/applications/networking/mailreaders/notmuch/default.nix:48]: Please use a commit id in the url, not just `master`

File: pkgs/applications/networking/mailreaders/notmuch/default.nix [hunk 2]
```diff
@@ -37,17 +43,39 @@ stdenv.mkDerivation rec {
...
+      hash = "sha256-Xs4hw3BPXBh8jGmx6UZsnuiosBCtjo8MWQlWAAgqKgg=";
+    })
+    (fetchpatch {
+      url = "https://raw.githubusercontent.com/void-linux/void-packages/master/srcpkgs/notmuch/patches/0005-configure-drop-gmime-session-key-extraction-support-.patch";
```
Comment (flokli) [inline pkgs/applications/networking/mailreaders/notmuch/default.nix:52]: Same

File: pkgs/applications/networking/mailreaders/notmuch/default.nix (hunk 1 above)
Comment (amjoseph-nixpkgs) [reply pkgs/applications/networking/mailreaders/notmuch/default.nix:48]: Yikes.

Comment (amjoseph-nixpkgs) [reply pkgs/applications/networking/mailreaders/notmuch/default.nix]: f95b944252e3a1fc900a18c8eeed59e6de9b6e9a

File: pkgs/applications/networking/mailreaders/notmuch/default.nix (hunk 2 above)
Comment (amjoseph-nixpkgs) [reply pkgs/applications/networking/mailreaders/notmuch/default.nix]: f95b944252e3a1fc900a18c8eeed59e6de9b6e9a

**Comment by amjoseph-nixpkgs** [comment]
Rebased.

File: pkgs/applications/networking/mailreaders/notmuch/default.nix [hunk 3]
```diff
@@ -49,12 +69,20 @@ stdenv.mkDerivation rec {
     # do not override CFLAGS of the Makefile created by mkmf
     substituteInPlace bindings/Makefile.local \
       --replace 'CFLAGS="$(CFLAGS) -pipe -fno-plt -fPIC"' ""
+    substituteInPlace configure \
+      --replace 'pkg-config' "${buildPackages.pkg-config}/bin/${buildPackages.pkg-config.targetPrefix}pkg-config"
+    substituteInPlace configure \
+      --replace './_check_x509_validity' 'true'
+    substituteInPlace configure \
+      --replace './_check_session_keys' 'true'
```
Comment (doronbehar) [inline pkgs/applications/networking/mailreaders/notmuch/default.nix:77]: Nit: You can do it all in a single `substituteInPlace` command.

File: pkgs/applications/networking/mailreaders/notmuch/default.nix [hunk 4]
```diff
@@ -40,6 +47,19 @@ stdenv.mkDerivation rec {
...
+      url = "https://raw.githubusercontent.com/void-linux/void-packages/8c1f637c706483577a22c8d4f2d06d5f041c9093/srcpkgs/notmuch/patches/0001-configure-detect-version-by-compiler-for-build.patch";
+      hash = "sha256-Xs4hw3BPXBh8jGmx6UZsnuiosBCtjo8MWQlWAAgqKgg=";
+    })
+    (fetchpatch {
+      url = "https://raw.githubusercontent.com/void-linux/void-packages/8c1f637c706483577a22c8d4f2d06d5f041c9093/srcpkgs/notmuch/patches/0005-configure-drop-gmime-session-key-extraction-support-.patch";
+      hash = "sha256-FXnNtkTqV7ZSIObQSwGu6THX0Rgvj4WeJySTIcJea1Y=";
+    })
+  ];
+
+  XAPIAN_CONFIG = "${xapian}/bin/xapian-config";
```
Comment (doronbehar) [inline pkgs/applications/networking/mailreaders/notmuch/default.nix:61]: This doesn't seem like it would behave well in a cross compilation...

File: pkgs/applications/networking/mailreaders/notmuch/default.nix [hunk 5]
```diff
@@ -40,6 +47,19 @@ stdenv.mkDerivation rec {
   ] ++ lib.optional withRuby ruby
     ++ lib.optional withSfsexp sfsexp;
 
+  patches = [
+    (fetchpatch {
```
Comment (doronbehar) [inline pkgs/applications/networking/mailreaders/notmuch/default.nix:51]: Too bad upstream uses mailing lists for tracking development... I know its a lot to ask, but are you aware of any attempts to tell them about these issues? It'd be nice to add some comment just in case.

File: pkgs/applications/networking/mailreaders/notmuch/default.nix [hunk 6]
```diff
@@ -28,6 +33,8 @@ stdenv.mkDerivation rec {
     pythonPackages.sphinx     # (optional) documentation -> doc/INSTALL
     texinfo                   # (optional) documentation -> doc/INSTALL
     pythonPackages.cffi
+    gnupg
```
Comment (doronbehar) [inline pkgs/applications/networking/mailreaders/notmuch/default.nix:36]: Are you sure the `gnupg` and `perl` in `buildInputs` are still needed?

File: pkgs/applications/networking/mailreaders/notmuch/default.nix [hunk 7]
```diff
@@ -124,6 +152,8 @@ stdenv.mkDerivation rec {
   + lib.optionalString withSfsexp ''
     cp notmuch-git $out/bin/notmuch-git
     wrapProgram $out/bin/notmuch-git --prefix PATH : $out/bin:${lib.getBin git}/bin
+  '' + ''
+    mkdir -p $info
```
Comment (doronbehar) [inline pkgs/applications/networking/mailreaders/notmuch/default.nix:156]: Why is this needed all of a sudden?

**Review by doronbehar — CHANGES_REQUESTED** [review]
Lastly, `https://github.com/NixOS/nixpkgs/pull/225111#discussion_r1160513273` is not a good commit message :)
